local.c

```diff
@@ -1221,9 +1221,16 @@ static int handle_protected_scan_element_attr(struct iio_channel *chn,
 
 	if (!strcmp(name, "index")) {
 		ret = local_read_dev_attr(dev, path, buf, sizeof(buf), false);
-		if (ret > 0)
-			chn->index = atol(buf);
+		if (ret > 0) {
+			char *end;
+			long long value;
+
+			value = strtoll(buf, &end, 0);
+			if (end == buf || value > LONG_MAX)
```

Comment:
Should we check if 'value' is negative? Since the value is assigned to chn->index and indexes aren't negative. Or they could be?

Reply:
No, they should always be equal to or greater than zero. I can add that before committing.

Reply:
should be fixed now.

Reply:
fixed